app/Providers/AuthServiceProvider.php

```diff
@@ -24,4 +26,18 @@ public function boot()
     {
         $this->registerPolicies();
     }
+
+    /**
+     * Register any auth-related application services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        if (config('auth.totp.bypass')) {
+            $this->app->bind(Google2FA::class, function ($app) {
+                return new Google2FAFake();
+            });
```

Comment:
Just a tiny nit, I believe we can bind without the closure:

```suggestion
            $this->app->bind(abstract: Google2FA::class, concrete: Google2FAFake::class);
```

Or if that's not to your liking, there's also shorthand functions now in PHP:
```suggestion
            $this->app->bind(Google2FA::class, fn () => new Google2FAFake());
```